libs/partners/huggingface/langchain_huggingface/chat_models/huggingface.py

```diff
@@ -40,6 +42,18 @@
 
 DEFAULT_SYSTEM_PROMPT = """You are a helpful, respectful, and honest assistant."""
 
+LLM_INFERENCE_CLIENT_MAP: Final[dict[str, str]] = {"max_new_tokens": "max_tokens"}
+LLM_INFERENCE_CLIENT_EXCLUDED_KWARGS: Final[set[str]] = {
```

Comment:
let's just remove these

Reply:
I think we can remove the `LLM_INFERENCE_CLIENT_EXCLUDED_KWARGS` without issues because we check the signature, as you noted in the other comment. 

The `LLM_INFERENCE_CLIENT_MAP` is there to override `max_new_tokens` -> `max_tokens` as the HF Endpoint is expecting `max_tokens` (while the default parameter for  is `max_new_tokens` in the class). See [HuggingFaceEndpoint._default_params](https://github.com/langchain-ai/langchain/blob/3048a9a26daa8c08a9a79543018708eb23f72c3f/libs/partners/huggingface/langchain_huggingface/llms/huggingface_endpoint.py#L255).

It seems like `max_tokens` is used for [/chat_completion](https://huggingface.co/docs/huggingface_hub/main/package_reference/inference_client#huggingface_hub.InferenceClient.chat_completion), while `max_new_tokens` is used for [/text_generation](https://huggingface.co/docs/huggingface_hub/main/package_reference/inference_client#huggingface_hub.AsyncInferenceClient.text_generation)

Overall this is why I said the implementation is crude, it seems to me like this should be handled inside the `transformers` library or `huggingface_hub` not in `langchain-huggingface`. Maybe this is an obscure artifact from Huggingface trying to comply with the OpenAI compatible web server, so now we have 2 different keyword arguments that are supposed to have identical functionality 🥴.

In addition they also have the `max_length` inside the transformers library, which is a third argument with (almost) the same purpose... 

```
max_length (int, optional, defaults to model.config.max_length) — The maximum length of the sequence to be generated.

max_new_tokens (int, optional, defaults to None) — The maximum numbers of tokens to generate, ignore the current number of tokens. Use either max_new_tokens or max_length but not both, they serve the same purpose.
```
[max_length/max_new_tokens](https://huggingface.co/docs/transformers/v4.18.0/en/main_classes/text_generation#transformers.generation_utils.GenerationMixin.generate.max_new_tokens)